crates/interfaces/src/consensus.rs

```diff
@@ -130,4 +130,39 @@ pub enum Error {
     WithdrawalIndexInvalid { got: u64, expected: u64 },
     #[error("Missing withdrawals")]
     BodyWithdrawalsMissing,
+    /// Thrown when calculating gas usage
+    #[error("gas uint64 overflow")]
+    GasUintOverflow,
+    /// returned if the transaction is specified to use less gas than required to start the
+    /// invocation.
+    #[error("intrinsic gas too low")]
+    GasTooLow,
+    /// returned if the transaction gas exceeds the limit
+    #[error("intrinsic gas too high")]
+    GasTooHigh,
+    /// thrown if a transaction is not supported in the current network configuration.
+    #[error("transaction type not supported")]
+    TxTypeNotSupported,
+    /// Thrown to ensure no one is able to specify a transaction with a tip higher than the total
+    /// fee cap.
+    #[error("max priority fee per gas higher than max fee per gas")]
+    TipAboveFeeCap,
```

Comment:
we already have these variants in rpc/error
as part of the transaction error.

so we should move that error type to interfaces instead I believe.

Reply:
> should we move this error type to consensus errors

I wouldn't classify them as consensus related.
so why are those variants added there and why can't we use the existing TransactionError?

> we have to inherit its error variants to consensus error

this does not sound right, if this is the case rn, this is not great.
`validate_transaction` should be able to distinguish be an actual transaction error or an error from the AccountProvider.

Reply:
> I wouldn't classify them as consensus related. so why are those variants added there and why can't we use the existing TransactionError?

We could use existing `InvalidTransactionError` with some variants added from `interfaces::consensus::Error` like `TransactionEip1559Disabled` instead.

Reply:
yes good idea, we need those!

Reply:
> > I wouldn't classify them as consensus related. so why are those variants added there and why can't we use the existing TransactionError?
> 
> We could use existing `InvalidTransactionError` with some variants added from `interfaces::consensus::Error` like `TransactionEip1559Disabled` .

I was wrong here. We can't use `InvalidTransactionError` in the crate `transaction-pool` due to cyclic imports.

Reply:
yeh, the InvalidTransactionError should be moved to interfaces I think

Reply:
I guess then we have to move some RPC errors such as `EthRpcErrorCode` to interfaces as well